users/config.rst

```diff
@@ -298,100 +305,141 @@ One or more ``folder`` elements must be present in the file. Each element
+.. option:: folder.device
+
+    These must have the ``id`` attribute and can have an ``introducedBy``
+    attribute, identifying the device that introduced us to share this folder
+    with the given device.  If the original introducer unshares this folder with
+    this device, our device will follow and unshare the folder (subject to
+    :opt:`skipIntroductionRemovals` being ``false`` on the introducer device).
+
     All mentioned devices are those that will be sharing the folder in question.
     Each mentioned device must have a separate ``device`` element later in the file.
     It is customary that the local device ID is included in all folders.
     Syncthing will currently add this automatically if it is not present in
     the configuration file.
 
-minDiskFree
-    The minimum required free space that should be available on the disk this folder
-    resides. The folder will be stopped when the value drops below the threshold. Accepted units are
-    ``%``, ``kB``, ``MB``, ``GB`` and ``TB``. Set to zero to disable.
+    The ``encryptionPassword`` sub-element contains the secret needed to decrypt
+    this folder's data on the remote device.  If left empty, the data is plainly
+    accessible (but still protected by the transport encryption).  The mechanism
+    and how to set it up is described in a :doc:`separate chapter <untrusted>`.
+
+.. option:: folder.minDiskFree
+
+    The minimum required free space that should be available on the disk this
+    folder resides.  The folder will be stopped when the value drops below the
+    threshold.  The element content is interpreted according to the given
+    ``unit`` attribute.  Accepted ``unit`` values are ``%`` (percent of the disk
+    / volume size), ``kB``, ``MB``, ``GB`` and ``TB``.  Set to zero to disable.
+
+.. option:: folder.versioning
 
-versioning
     Specifies a versioning configuration.
 
-.. seealso::
-    :ref:`versioning`
+    .. seealso::
+	:doc:`versioning`
+
+.. option:: folder.copiers
+	    folder.hashers
 
-copiers, pullers, hashers
-    The number of copier, puller and hasher routines to use, or ``0`` for the
+    The number of copier and hasher routines to use, or ``0`` for the
     system determined optimums. These are low level performance options for
     advanced users only; do not change unless requested to or you've actually
     read and understood the code yourself. :)
 
-order
+.. option:: folder.pullerMaxPendingKiB
+
+    Controls when we stop sending requests to other devices once we’ve got this
+    much unserved requests.  The number of pullers is automatically adjusted
+    based on this desired amount of outstanding request data.
+
+.. option:: folder.order
+
     The order in which needed files should be pulled from the cluster.
     The possibles values are:
 
-    random
+    ``random`` (default)
         Pull files in random order. This optimizes for balancing resources among
         the devices in a cluster.
```

Comment:
Same as above, but I'm not sure what to change "cluster" to here. Maybe "among the connected devices" or "among the devices sharing the same folder"?

Reply:
What's wrong with the word "cluster"?  It's technically correct, though not a technical term: "a collection of things somehow tied together".  If there's no obvious better term, let's just leave it at that.

Note that we're looking at the detailed documentation for an XML file format here.  As such we don't necessarily need to aim for Average Joe to understand everything.

Reply:
I treat "cluster" similarly to "node" and "repository". All are present in the English dictionary, but in reality they're not used by the general population (when it comes to computers and such). As for Syncthing, "nodes" were replaced by "devices", "repositories" by "folders" a few years ago. "Cluster" was left out in the process.

I just think it's beneficial to use simpler terms unless there is no other option. While the average Joe is unlikely to come and explore the docs on his own, he is very often sent here from the forum, so the easier to understand the explanations are, the better.

Reply:
Well, "node" is closely related to graphs and "repository" is technical, both rather uncommon words.  

> "Cluster" was left out in the process.

And I think it's still ok to use.

Reply:
"Cluster", at least regarding computing, is on the same level as the other two, really. I mean, there's a reason why cloud/sync solutions aimed at the general public (e.g. Dropbox, Google Drive, OneDrive, etc.) don't use such terminology in their documentation, unless it's from the get-go very clearly directed strictly at developers and/or power users.

I'm not saying this needs to be addressed in this commit, but the question remains whether the documentation (and wording in general) should be accessible to all users of Syncthing, including non-technical people, or only targeted at techies and such. I'd also keep in mind that we're in the general "usage" section of the Docs here, not "development" or even "advanced".

Reply:
1. Dropbox etc. use a client-server model, where "cluster" is not as meaningful as for a decentralized system.
2. This page explains how to interpret entries in an XML file.  Possibly will get linked to from the "advanced" configuration panel.  It's directly strictly at developers and power users, otherwise we should definitely get rid of the large XML quotes :wink:

Glad to hear you don't insist on this discussion being finished within this PR.

Reply:
> Dropbox etc. use a client-server model

Dropbox is a little more complex than that, e.g. it also supports direct sync between LAN devices without downloading files from the central server, but let's not go into such details here.

> This page explains how to interpret entries in an XML file.  Possibly will get linked to from the "advanced" configuration panel.

Yeah, this is a different problem, I'd say mainly about a discrepancy between the Web GUI and the Docs. The two are structured differently, and what's clearly divided in the GUI between "basic" and "advanced" configuration, in the Docs is all simply presented at the same page. As such, it becomes the source of information for both power users and beginners. I don't think there's an easy solution to this problem, but I personally just always prefer to stick to using language that is easy and understandable by everyone (which includes beginners, of course, but also non-native English speakers, especially since the Docs are not available in other languages yet).

Reply:
Actually there are a few help links from the regular "Settings" dialog which lead directly to the detailed sub-pages here (e. g. GUI listen address, Automatic upgrades).  But others link to the config description, which could be improved.  There simply isn't a "novice-compatible" help topic for everything.  Feel free to write up more stuff in more easily understood language and submit it.  The distinction itself makes sense to me, because no regular user wants to scroll through XML code.